jupyterlab_server/handlers.py

```diff
@@ -179,7 +179,7 @@ class NotFoundHandler(LabHandler):
     @lru_cache  # noqa: B019
     def get_page_config(self) -> dict[str, Any]:
         """Get the page config."""
-        page_config = super().get_page_config()
+        page_config = super().get_page_config().copy()
```

Comment:
Can you leave a top level comment on why this fixes the issue?

Reply:
Is the idea that we don't want to set `notFoundUrl` on all subsequent requests, just this one?